bungee/src/main/java/com/github/games647/commandforward/bungee/CommandForwardBungee.java

```diff
@@ -41,49 +38,49 @@ public void onServerConnected(PluginMessageEvent messageEvent) {
 
         //check if the message is sent from the server
         if (Server.class.isAssignableFrom(messageEvent.getSender().getClass())) {
-            parseMessage((CommandSender) messageEvent.getReceiver(), ByteStreams.newDataInput(messageEvent.getData()));
+            parseMessage((CommandSender) messageEvent.getReceiver(),
+                ByteStreams.newDataInput(messageEvent.getData()));
         }
     }
 
     private void parseMessage(CommandSender sender, ByteArrayDataInput dataInput) {
-        boolean isPlayer = dataInput.readBoolean();
-        String command = dataInput.readUTF();
-        String arguments = dataInput.readUTF();
-
-        CommandSender invoker = getProxy().getConsole();
-        if (isPlayer) {
-            invoker = sender;
-        }
+        final boolean isPlayer = dataInput.readBoolean();
+        final String command = dataInput.readUTF();
+        final String arguments = dataInput.readUTF();
+        final CommandSender invoker = (isPlayer) ? sender : getProxy().getConsole();
 
         invokeCommand(invoker, dataInput.readBoolean(), command, arguments);
     }
 
     private void invokeCommand(CommandSender invoker, boolean isOp, String command, String arguments) {
         PluginManager pluginManager = getProxy().getPluginManager();
-        if (isOp) {
-            try {
-                Map<String, Command> commandMap = (Map<String, Command>) pluginManager.getClass()
-                        .getField("commandMap").get(pluginManager);
 
-                Command pluginCmd = commandMap.get(command);
-                if (pluginCmd == null) {
-                    invoker.sendMessage(new ComponentBuilder("Command not known")
-                            .color(ChatColor.RED)
-                            .create());
-                } else {
+        if (isOp) {
+            pluginManager.getCommands()
+                .stream()
+                .filter(entry -> entry.getKey().equals(command.toLowerCase())).findFirst()
+                .map(Map.Entry::getValue)
+                .map(pluginCmd -> {
                     pluginCmd.execute(invoker, arguments.split(" "));
-                }
-            } catch (NoSuchFieldException | IllegalAccessException ex) {
-                String exMess = ex.getMessage();
-                BaseComponent[] message = new ComponentBuilder("Error occurred executing command " + exMess)
-                        .color(ChatColor.RED)
-                        .create();
-                invoker.sendMessage(message);
-                getLogger().log(Level.WARNING, "Cannot access command map for executing command", ex);
-            }
+                    return pluginCmd;
```

Comment:
It's a little weird to use mapping for consuming here invoking the command. In Java 9, it would be great to use `ifPresentOrElse`. However since we are here targeting Java 8, I'm fine with it.

Reply:
Sorry I don't have a better solution for Java 8 in mind.